src/Microsoft.AspNetCore.Hosting/Internal/HostingEventSource.cs

```diff
@@ -0,0 +1,94 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Diagnostics.Tracing;
+using Microsoft.AspNetCore.Http;
+
+namespace Microsoft.AspNetCore.Hosting.Internal
+{
+    [EventSource(Name = "Microsoft-AspNetCore-Hosting")]
+    public sealed class HostingEventSource : EventSource
+    {
+        public static readonly HostingEventSource Log = new HostingEventSource();
+
+        private HostingEventSource() { }
+
+        [Event(1, Level = EventLevel.Informational)]
+        public void HostStart()
+        {
+            WriteEvent(1);
+        }
+
+        [Event(2, Level = EventLevel.Informational)]
+        public void HostStop()
+        {
+            WriteEvent(2);
+        }
+
+        [NonEvent]
+        public void RequestStart(HttpContext context)
+        {
+            if (IsEnabled())
+            {
+                RequestStart(
+                    context.TraceIdentifier,
+                    context.Request.Protocol,
+                    context.Request.Method,
+                    context.Request.ContentType ?? string.Empty,
+                    context.Request.ContentLength.HasValue ? context.Request.ContentLength.Value.ToString() : string.Empty,
+                    context.Request.Scheme,
+                    context.Request.Host.ToString(),
+                    context.Request.PathBase,
+                    context.Request.Path,
+                    context.Request.QueryString.ToString());
+            }
+        }
+
+        [NonEvent]
+        public void RequestStop(HttpContext context, Exception exception = null)
+        {
+            if (IsEnabled())
+            {
+                RequestStop(
```

Comment:
What do you think about adding all fields from RequestStart to RequestStop as well?

Let's say I want to log each request to an external system (file system, Azure Log analytics, ...) and I want to be able to do reports on request durations per URL. 

Right now, the logger has to subscribe to both events and send both to the external system. This results in more traffic and data overhead and the external system must be able to merge the two events to do the reporting.
Alternatively, the logger could do the merging of the RequestStart/RequestStop events before it sends them but this also results in some complexity overhead (the RequestStart-event must be cached, etc if I'm correct).

Having all fields on RequestStop would allow these loggers to subscribe to just one event which contains all necessary information.
Systems that want to subscribe to both events are free to ignore these fields from RequestStop.